src/utils/utils.hpp

```diff
@@ -17,6 +17,20 @@
 #include "utils/optional.hpp"
 #include "utils/result.hpp"
 
+// compatible for c++11 but GCC < 5
+#if defined(__GNUC__) && (__GNUC__ < 5) && !defined(__clang__)
+#if 201103L <= __cplusplus
+namespace std
+{
```

Comment:
I'm not comfortable adding to `namespace std` like this in the library, even for internal code guarded on a specific compiler version. I think a better fix here would be to just use `std::is_trivial` instead of `std::is_trivially_copyable`, the later of which is preferable since `std::is_trivial` was deprecated but it'll work for the older gcc version.